File: examples/scrape-glassdoor/rest/java/ScrapeGlassdoor.java
// Scrapes Glassdoor job listings using BQL with stealth mode and residential proxy.
//
// Run: javac ScrapeGlassdoor.java && java ScrapeGlassdoor

import java.net.URI;
import java.net.http.*;

public class ScrapeGlassdoor {
    public static void main(String[] args) throws Exception {
        String token = "YOUR_API_TOKEN_HERE";
        String endpoint = "https://production-sfo.browserless.io/stealth/bql?token="
            + token + "&proxy=residential&proxyCountry=us";

        String query = "mutation ScrapeGlassdoor {"
            + " goto(url: \\\"https://www.glassdoor.com/Job/new-york-software-engineer-jobs-SRCH_IL.0,8_IC1132348_KO9,26.htm\\\", waitUntil: networkIdle) { status }"
            + " jobs: mapSelector(selector: \\\"[data-test='jobListing']\\\") {"
            + "   title: mapSelector(selector: \\\"a[data-test='job-title']\\\") { innerText }"
            + "   company: mapSelector(selector: \\\"[data-test='employer-name']\\\") { innerText }"
            + "   location: mapSelector(selector: \\\"[data-test='emp-location']\\\") { innerText }"
            + "   salary: mapSelector(selector: \\\"[data-test='detailSalary']\\\") { innerText }"
            + " }"
            + " }";

        String payload = "{\"query\": \"" + query + "\", \"variables\": {}, \"operationName\": \"ScrapeGlassdoor\"}";

        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
            .uri(URI.create(endpoint))
            .header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString(payload))
            .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
        System.out.println(response.body());
    }
}

File: examples/scrape-glassdoor/rest/python/scrape_glassdoor.py
# Scrapes Glassdoor job listings using BQL with stealth mode and residential proxy.
#
# Install: pip install requests
# Run:     python scrape_glassdoor.py

import requests

query = """
mutation ScrapeGlassdoor {
  goto(
    url: "https://www.glassdoor.com/Job/new-york-software-engineer-jobs-SRCH_IL.0,8_IC1132348_KO9,26.htm"
    waitUntil: networkIdle
  ) {
    status
  }
  jobs: mapSelector(selector: "[data-test='jobListing']") {
    title: mapSelector(selector: "a[data-test='job-title']") {
      innerText
    }
    company: mapSelector(selector: "[data-test='employer-name']") {
      innerText
    }
    location: mapSelector(selector: "[data-test='emp-location']") {
      innerText
    }
    salary: mapSelector(selector: "[data-test='detailSalary']") {
      innerText
    }
  }
}
"""

response = requests.post(
    'https://production-sfo.browserless.io/stealth/bql',
    params={
        'token': 'YOUR_API_TOKEN_HERE',
        'proxy': 'residential',
        'proxyCountry': 'us',
    },
    json={'query': query, 'variables': {}, 'operationName': 'ScrapeGlassdoor'},
)

data = response.json()['data']
for job in data['jobs']:
    title = job['title'][0]['innerText'] if job['title'] else ''
    company = job['company'][0]['innerText'] if job['company'] else ''
    location = job['location'][0]['innerText'] if job['location'] else ''
    salary = job['salary'][0]['innerText'] if job['salary'] else ''
    print(f'{title} at {company} — {location} — {salary}')
